lib/WebService/GoogleAPI/Client.pm

```diff
@@ -269,26 +269,22 @@ sub api_query
 
       foreach my $meth_param_spec ( keys %{ $method_discovery_struct->{ parameters } } )
       {
+        carp (pp $params) if $self->debug > 10;
         carp("Checking $meth_param_spec") if ($self->{debug}>10);
-        ## set default value if is not provided within $params->{options} - nb technically not required but provides visibility of the params if examining the options when debugging
-        $params->{ options }{ $meth_param_spec } = $method_discovery_struct->{ parameters }{ $meth_param_spec }{ default }
-          if (
-          ( not defined $params->{ options }{ $meth_param_spec } )
-          && ( defined $method_discovery_struct->{ parameters }{ $meth_param_spec }{ default }
-            && $method_discovery_struct->{ parameters }{ $meth_param_spec }{ location } eq 'query' )
-          );
-        ## this looks to be clobbering all options - TODO - review and stop clobbering if already defined
+	#Default values are set inside of conditional later
+	
+	
+	## this looks to be clobbering all options - 
+	#TODO - review and stop clobbering if already defined
 
-        carp( "checking discovery spec'd parameter - $meth_param_spec" ) if $self->debug > 10;
 
         #carp("$meth_param_spec  has a user option value defined") if ( defined $params->{options}{$meth_param_spec} );
-        if ( $params->{ path } =~ /\{.+\}/xms )    ## there are un-interpolated variables in the path - try to fill them for this param if reqd
+          if ( $method_discovery_struct->{ parameters }{ $meth_param_spec }{ 'location' } eq 'path' )    ## this is a path variable
```

Comment:
these conditionals were stack incorrectly, and it only looked at query params if there was something in the path that needed looking at.